### PR TITLE
v.checker: disallow assigning none to _

### LINKED ISSUE #20645: cgen error for ` _ = none`
```bash
v run a.v
```
```
==================
/tmp/v_2625/a.01HMWV3D4QDHES1C1B00P00B2E.tmp.c:12858: error: '{' expected (got ";")
...
==================
(Use `v -cg` to print the entire error message)                    
builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```
```bash
cat a.v
```
```v
fn x() ?bool {
        return none
}                                                                  
_ = x() or {
        _ = none
        true
}
```
I have no idea what's happening here, I was experimenting as I'm still learning the language and then it asked me to report.

and I want to leverage this opportunity to ask an off-topic little doubt,
in structs, what `__global:` allows that `pub mut:` doesn't?
I tried them in various forms and both worked fine,
the docs talks about "parent module only" stuff with `pub mut:` but I don't get it they both work no matter what I try.
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

### PR DESCRIPTION
**Pull request by pierrec**
Fixes #20645.

### CODE REVIEW

File: vlib/v/checker/tests/assign_blank_none_err.out [hunk 1]
```diff
@@ -0,0 +1,5 @@
+vlib/v/checker/tests/assign_blank_none_err.vv:2:2: error: cannot assign none to blank `_` identifier
+    1 | fn main() {
+    2 |     _ = none
+      |     ^
+    3 | }
```
Comment (Delta456) [inline vlib/v/checker/tests/assign_blank_none_err.out:5]: Add new line at the end

Comment (pierrec) [reply vlib/v/checker/tests/assign_blank_none_err.out:5]: @Delta456 why is a newline required here?

**Review by shove70 — CHANGES_REQUESTED** [review]

**Comment by Delta456** [comment]
Run `v fmt -w vlib/v/checker/assign.v`

**Review by shove70 — APPROVED** [review]

File: vlib/v/checker/tests/assign_blank_none_err.out (hunk 1 above)
Comment (JalonSolov) [reply vlib/v/checker/tests/assign_blank_none_err.out]: It's only needed in a .out file if it helps match against the test output.

**Pull request merged**